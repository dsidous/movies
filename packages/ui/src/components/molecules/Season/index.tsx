import React from 'react';
import { Link } from 'react-router-dom';
import { Typography } from '@material-ui/core';

import { MediaImage } from '@atoms/index';
import useStyles from './styles';
import { Season } from '@typesRoots/tv';

interface Props {
  season: Season;
  tvId: number;
}

const Season: React.FC<Props> = ({
  season: {
    poster_path,
    overview,
    name,
    episode_count,
    air_date,
    season_number,
  },
  tvId,
}) => {
  const classes = useStyles();

  const date = air_date !== null ? air_date.slice(0, 4) : '';
  const link = `/tv/${tvId}/season/${season_number}`;

  return (
    <div className={classes.root}>
      <Link to={link} className={classes.poster}>
        <MediaImage
          mediaType='poster'
          size={3}
          filePath={poster_path}
          name={name}
        />
      </Link>
      <div className={classes.details}>
        <Typography variant='h6'>{name}</Typography>
        <Typography
          paragraph
        >{`${date} | ${episode_count} episodes`}</Typography>
        <Typography variant='body2'>{overview}</Typography>
      </div>
    </div>
  );
};

export default Season;
