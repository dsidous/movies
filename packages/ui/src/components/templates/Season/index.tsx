import React from 'react';

import Episode from '../../molecules/Episode';
import PageTransition from '../../atoms/PageTransition';
import MiniHeader from '../../molecules/MiniHeader';
import useStyles from './styles';
import Skeleton from './skeleton';
import { Tv_Season } from '@typesRoots/tvseasons';

interface Props {
  loading: boolean;
  tvSeason: Tv_Season;
  tvId: string;
}

const Season: React.FC<Props> = ({
  loading,
  tvSeason: { name, air_date, episodes, poster_path },
  tvId,
}) => {
  const classes = useStyles();

  if (loading) {
    return <Skeleton />;
  }

  return (
    <PageTransition>
      <MiniHeader
        title={name}
        release_date={air_date}
        poster_path={poster_path}
        link={`/tv/${tvId}/seasons`}
        linkCopy='Back to seasons'
      />

      <ul className={classes.season__wrapper}>
        {episodes.map((episode) => (
          <li className={classes.season__element} key={episode.episode_number}>
            <Episode episode={episode} />
          </li>
        ))}
      </ul>
    </PageTransition>
  );
};

export default Season;
