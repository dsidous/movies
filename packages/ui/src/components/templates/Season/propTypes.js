import { object, number, bool } from 'prop-types';

export const propTypes = {
  loading: bool,
  tvSeason: object,
  tvId: number,
};

export const defaultProps = {};
