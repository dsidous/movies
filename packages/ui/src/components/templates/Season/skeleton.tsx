import React from 'react';
import Skeleton from '@material-ui/lab/Skeleton';
import { Container, Typography } from '@material-ui/core';

import useStyles from './styles';

export default () => {
  const classes = useStyles();
  return (
    <Container>
      <Container className={classes.header}>
        <Skeleton variant='rect' width={50} height={80} />
        <Container>
          <Skeleton width='30%' />
          <Skeleton width='20%' />
        </Container>
      </Container>
      <ul className={classes.list}>
        {Array.from(new Array(4)).map((_, index) => (
          <li className={classes.item} key={index}>
            <Skeleton variant='rect' width={300} height={80} />
            <Container>
              <Typography variant='h6'>
                <Skeleton variant='text' width='30%' />
              </Typography>
              <Skeleton />
              <Skeleton />
              <Skeleton width='60%' />
            </Container>
          </li>
        ))}
      </ul>
    </Container>
  );
};
