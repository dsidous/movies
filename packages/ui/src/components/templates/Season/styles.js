import { makeStyles } from '@material-ui/core/styles';

export default makeStyles(theme => ({
  season__wrapper: {
    margin: '0 auto',
    padding: '0 20px',
  },

  season__element: {
    borderBottom: theme.separator,
    listStyleType: 'none',
    margin: '40px 0',
    padding: '0 0 40px',
  },
}));
