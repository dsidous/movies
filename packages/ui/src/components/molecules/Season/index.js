/* eslint-disable camelcase */
import React from 'react';
import { Link } from 'react-router-dom';

import { propTypes } from './propTypes';
import MediaImage from '../../atoms/MediaImage';

const Season = props => {
  const {
    season: {
      poster_path,
      overview,
      name,
      episode_count,
      air_date,
      season_number,
    },
    tvId,
  } = props;

  const date = air_date !== null ? air_date.slice(0, 4) : '';
  const link = `/tv/${tvId}/season/${season_number}`;

  return (
    <div className="season">
      <Link to={link} className="season-poster">
        <MediaImage
          mediaType="poster"
          size={3}
          filePath={poster_path}
          name={name}
        />
      </Link>
      <div className="season-details">
        <h3>{name}</h3>
        <h5>{`${date} | ${episode_count} episodes`}</h5>
        <div>{overview}</div>
      </div>
    </div>
  );
};

Season.propTypes = propTypes;

export default Season;
