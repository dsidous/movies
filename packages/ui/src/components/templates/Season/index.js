/* eslint-disable camelcase */
import React from 'react';

import { propTypes } from './propTypes';
import Spinner from '../../atoms/Spinner';
import Episode from '../../molecules/Episode';
import PageTransition from '../../atoms/PageTransition';
import MiniHeader from '../../molecules/MiniHeader';

const Season = props => {
  const { loading } = props;

  if (loading) {
    return <Spinner />;
  }

  const {
    tvSeason: { name, air_date, episodes, poster_path },
    tvId,
  } = props;

  return (
    <PageTransition>
      <MiniHeader
        title={name}
        release_date={air_date}
        poster_path={poster_path}
        link={`/tv/${tvId}/seasons`}
        linkCopy="Back to seasons"
      />

      <ul className="episodes__wrapper">
        {episodes.map(episode => (
          <li className="episodes__element" key={episode.episode_number}>
            <Episode episode={episode} />
          </li>
        ))}
      </ul>
    </PageTransition>
  );
};

Season.propTypes = propTypes;

export default Season;
