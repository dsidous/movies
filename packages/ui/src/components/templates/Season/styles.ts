import { makeStyles } from '@material-ui/core/styles';

export default makeStyles((theme) => ({
  header: {
    alignItems: 'center',
    borderBottom: theme.separatorBold,
    display: 'flex',
    padding: theme.spacing(4, 0),
  },

  list: {
    padding: 0,
  },

  item: {
    borderBottom: theme.separator,
    display: 'flex',
    justifyContent: 'space-between',
    marginBottom: theme.spacing(2),
    paddingBottom: theme.spacing(2),

    '& h6': {
      marginBottom: theme.spacing(2),
    },
  },
  season__wrapper: {
    margin: '0 auto',
    padding: theme.spacing(0, 3),
  },

  season__element: {
    borderBottom: theme.separator,
    listStyleType: 'none',
    margin: theme.spacing(5, 0),
    padding: theme.spacing(0, 0, 5),
  },
}));
