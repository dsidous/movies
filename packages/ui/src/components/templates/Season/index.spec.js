import React from 'react';
import { shallow } from 'enzyme';
import Season from '.';

const mockProps = {
  tvSeason: {
    name: 'name',
    air_date: '01.01.2010',
    episodes: [
      {
        episode_number: 1,
      },
    ],
    poster_path: 'path/to/image',
  },
  tvId: 1,
  loading: false,
};

describe('Templates/Season', () => {
  it('should render as expected', () => {
    const wrapper = shallow(<Season {...mockProps} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should render a spinner when loading expected', () => {
    const wrapper = shallow(<Season {...mockProps} loading />);

    expect(wrapper).toMatchSnapshot();
  });
});
