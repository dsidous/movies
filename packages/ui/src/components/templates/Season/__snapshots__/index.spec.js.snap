// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Templates/Season should render a spinner when loading expected 1`] = `ShallowWrapper {}`;

exports[`Templates/Season should render as expected 1`] = `ShallowWrapper {}`;
